source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/users/list/__tests__/filter-by-organization-acceptance-test.tsx

```diff
@@ -0,0 +1,140 @@
+import switcher from '@ui/components/sidebar/org-switcher/__tests__/page';
+import page from './page';
+
+describe('Acceptance | User list | Filtering users by organization', () => {
+
+  setupApplicationTest();
+  setupRequestInterceptor();
+
+  describe('User belongs multiple organizations', () => {
+
+    useFakeAuthentication(undefined, {
+      data: [{
+        type: 'organizations', id: 1,
+        attributes: {
+          name: 'SIL International'
+        }
+      }, {
+        type: 'organizations', id: 2,
+        attributes: {
+          name: 'DeveloperTown'
+        }
+      }, {
+        type: 'organizations', id: 3,
+        attributes: {
+          name: 'Kalaam Media'
+        }
+      }]
+    });
+
+    beforeEach(function() {
+      this.mockGet(200, '/groups', {
+        data: [{
+          type: 'groups',
+          id: '2',
+          attributes: {
+            name: 'Fake group'
+          }
+        }]
+      });
+    });
+
+    beforeEach(function () {
+      this.mockGet(200, '/users', {
+        data: [{
+          type: 'users',
+          id: 2,
+          attributes: { id: 1, familyName: 'fake', givenName: 'One' },
+          relationships: {
+            ['organization-memberships']: { data: [ { id: 2, type: 'organization-memberships' }]},
+            "group-memberships": {}
+          }
+        }, {
+          type: 'users',
+          id: 3,
+          attributes: { id: 1, familyName: 'fake', givenName: 'Two' },
+          relationships: {
+            ['organization-memberships']: { data: [ { id: 3, type: 'organization-memberships' }]},
+            "group-memberships": {}
+          },
+        }],
+        included: [
+          {
+            id: 2, type: 'organization-memberships',
+            attributes: {},
+            relationships: {
+              user: { data: { id: 2, type: 'users' } },
+              organization: { data: { id: 1, type: 'organizations' } }
+            }
+          },
+          {
+            id: 3, type: 'organization-memberships',
+            attributes: {},
+            relationships: {
+              user: { data: { id: 3, type: 'users' } },
+              organization: { data: { id: 2, type: 'organizations' } }
+            }
+          }
+        ]
+      });
+    });
+
+    beforeEach(async function () {
+      await visit('/users');
+    });
+
+    describe('Render user table when "all organization" is selected',() => {
+
+      beforeEach(async function () {
+        await app.openSidebar();
+        await app.openOrgSwitcher();
+        await switcher.selectAllOrg();
+      });
+
+      it('all user are rendered', () => {
+
+        expect(page.usernames().length).to.equal(3);
+
+        const usernames = page.usernames();
+        const text = usernames.map(u => u.text).join();
+
+        expect(text).to.include('fake fake');
+        expect(text).to.include('One fake');
+        expect(text).to.include('Two fake');
+      });
```

Comment:
there should be a nested test in here for switching the organization after already viewing the users screen, and making sure users stop showing, because that was the bug :)

Reply:
The current set of tests only test what happens if you start off by visiting the users page with only a particular org selected. Gotta test the switch :)

Reply:
Isnt the next test testing that? The first one test that you see the page then i change the organization and I get New result. Perhaps a test that test the negative path that if I change the organization it wont add up?

Reply:
according to the (lack of) nesting of the describe blocks, they are separate scenarios and are independent of each other.

This is the flow that needs to exist as a test:
 - Select All Organizations
    - View Users Page
       -> should see all users
    - Select a specific Organization (still on the users page)
       -> should see a subset of the previously asserted users